src/server/admin/models.py

```diff
@@ -35,97 +35,85 @@ def __init__(self, api_key: str, email: str = None) -> None:
+                  session,
+                  user_id: Optional[int] = None, api_key: Optional[str] = None, user_email: Optional[str] = None
     ) -> "User":
         # NOTE: be careful, using multiple arguments could match multiple users, but this will return only one!
-        with Session() as session:
-            user = (
-                session.query(User)
-                .filter((User.id == user_id) | (User.api_key == api_key) | (User.email == user_email))
-                .first()
-            )
+        user = (
+            session.query(User)
+            .filter((User.id == user_id) | (User.api_key == api_key) | (User.email == user_email))
+            .first()
+        )
         return user if user else None
 
     @staticmethod
-    def create_user(api_key: str, email: str, user_roles: Optional[Set[str]] = None) -> "User":
+    @default_session(WriteSession)
+    def create_user(api_key: str, email: str, session, user_roles: Optional[Set[str]] = None) -> "User":
         get_structured_logger("api_user_models").info("creating user", api_key=api_key)
-        with Session() as session:
-            new_user = User(api_key=api_key, email=email)
-            # TODO: we may need to populate 'created' field/column here, if the default
-            #   specified above gets bound to the time of when that line of python was evaluated.
-            session.add(new_user)
-            session.commit()
-            User._assign_roles(new_user, user_roles, session)
-            session.commit()
-        return new_user
+        new_user = User(api_key=api_key, email=email)
+        session.add(new_user)
+        session.commit()
+        return User._assign_roles(new_user, user_roles, session)
 
     @staticmethod
+    @default_session(WriteSession)
     def update_user(
         user: "User",
         email: Optional[str],
         api_key: Optional[str],
-        roles: Optional[Set[str]]
+        roles: Optional[Set[str]],
+        session
     ) -> "User":
         get_structured_logger("api_user_models").info("updating user", user_id=user.id, new_api_key=api_key)
-        with Session() as session:
-            user = User.find_user(user_id=user.id)
-            if user:
-                update_stmt = (
-                    update(User)
-                    .where(User.id == user.id)
-                    .values(api_key=api_key, email=email)
-                )
-                session.execute(update_stmt)
-                User._assign_roles(user, roles, session)
-                session.commit()
-        return user
+        user = User.find_user(user_id=user.id, session=session)
+        if user:
+            update_stmt = (
+                update(User)
+                .where(User.id == user.id)
+                .values(api_key=api_key, email=email)
+            )
+            session.execute(update_stmt)
+            return User._assign_roles(user, roles, session)
+        # TODO: else: raise Exception() ??
```

Comment:
¯\\\_(ツ)_/¯ im ok with neither because our usage of this seems consistent enough right now.  a future caller of this can get the same information by checking for a `None` return value instead of catching an exception.  but its easy enough to do, so ill just do it.  change incoming shortly...